UnitySDK/Assets/ML-Agents/Scripts/Sensor/WriteAdapter.cs

```diff
@@ -0,0 +1,105 @@
+using System.Collections.Generic;
+using MLAgents.InferenceBrain;
+
+namespace MLAgents.Sensor
+{
+    /// <summary>
+    /// Allows sensors to write to both TensorProxy and float arrays/lists.
+    /// </summary>
+    public class WriteAdapter
```

Comment:
Not sure about how I feel about this write adapter. Is this going to be user facing ? Is this what we want the API to look like ?

Reply:
It's not user facing, unless they implement a custom ISensor, and even then, they only need to write to it, and even _then_ they can use SensorBase to avoid it altogether.

I took this approach instead of trying to use ArraySegment (which might have had it's own problems) because writes to Barracuda.Tensor have some additional sideeffects (calls PrepareCacheForAccess and sets a dirty flag); I couldn't see another suitable way to abstract the writes.

Reply:
Is there a better way to hide it or mark it as "don't use"?

Reply:
(talked offline, Vince is OK with this)